pandas/core/dtypes/base.py

```diff
@@ -0,0 +1,122 @@
+"""Extend pandas with custom array types"""
+import abc
+
+from pandas.compat import add_metaclass
+
+
+@add_metaclass(abc.ABCMeta)
+class ExtensionDtype(object):
+    """A custom data type for your array.
+    """
+
+    def __repr__(self):
+        return str(self)
```

Comment:
Something like `return "{}()".format(self.__class__.__name__)` ?

Or we could also remove this. The default will be just a longer version from what I propose.

Reply:
Perhaps remove for now, and do something like scikit-learn does with building the repr from `__init__`'s signature in the future?